lib/upload.js

```diff
@@ -155,6 +155,23 @@ class BaseUpload {
       })
   }
 
+  static convertToStringHeaderValue(val) {
```

Comment:
The helper method is a good idea, but not really needed. I would like to avoid a large helper method like this that magically converts everything somehow into a string and instead use dedicated methods where we need to convert things into string.